plugins/azure/virtualmachines/desiredSkuSize.js

```diff
@@ -0,0 +1,66 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'VM Desired SKU Size',
+    category: 'Virtual Machines',
+    description: 'Ensures that virtual machines is using the desired SKU size.',
+    more_info: 'VM SKU size defines the compute power and data processing speed. VM SKU size should be chosen carefully to address compute requirements for the organization and to save un-necessary costs.',
+    recommended_action: 'Resize VM to desired SKU size.',
+    link: 'https://docs.microsoft.com/en-us/azure/virtual-machines/sizes',
+    apis: ['virtualMachines:listAll'],
+    settings: {
+        vm_desired_sku_size: {
+            name: 'VM Desired SKU Size',
+            description: 'SKU size for the virtual machines. Created virtual machine SKU sizes should match the desired SKU size',
+            regex: '/(^Basic_A(.*?)$)|(^Standard_(A|B|D|E|F|G|H|L|M|N)(.*?)$)/i',
+            default: 'standard_ds3_v2'
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        const config = {
+            desiredSkuSize: settings.vm_desired_sku_size || this.settings.vm_desired_sku_size.default
+        };
+
```

Comment:
`if (!config.desiredSkuSize.elngth) return callback(null, results, source);`

Reply:
This should be an opt-in plugin

Reply:
Also make this setting a comma-separated list so that they can choose from a list